qp_shotgun/kneaddata/kneaddata.py

```diff
@@ -0,0 +1,287 @@
+    samples have both a forward and a reverse sequence. However, the read
+    trimming step can sometimes eliminate all reverse reads, especially in low
+    coverage samples with poor overall reverse read quality.
+    """
+
+    # sort forward seqs
+    forward_seqs.sort()
+
+    # check that rev seqs are same len
+    if reverse_seqs:
+        if len(forward_seqs) != len(reverse_seqs):
+            raise ValueError('Your reverse and forward files are of different '
+                             'length. Forward: %s. Reverse: %s.' %
+                             (', '.join(forward_seqs),
+                              ', '.join(reverse_seqs)))
+        reverse_seqs.sort()
+
+    # get run prefixes
+    # These are prefixes that should match uniquely to forward reads
+    # sn_by_rp is dict of samples keyed by run prefixes
+    sn_by_rp = get_sample_names_by_run_prefix(map_file)
+
+    # make pairings
+    samples = []
+    used_prefixes = set()
+    for i, (fwd_fp, rev_fp) in enumerate(zip_longest(forward_seqs,
+                                                     reverse_seqs)):
+        # fwd_fp is the fwd read filepath
+        fwd_fn = basename(fwd_fp)
+
+        # iterate over run prefixes and make sure only one matches
+        run_prefix = None
+        for rp in sn_by_rp:
+            if fwd_fn.startswith(rp) and run_prefix is None:
+                run_prefix = rp
+            elif fwd_fn.startswith(rp) and run_prefix is not None:
+                raise ValueError('Multiple run prefixes match this fwd read: '
+                                 '%s' % fwd_fn)
+
+        # make sure that we got one matching run prefix:
+        if run_prefix is None:
+            raise ValueError('No run prefix matching this fwd read: %s'
+                             % fwd_fn)
+
+        if run_prefix in used_prefixes:
+            raise ValueError('This run prefix matches multiple fwd reads: '
+                             '%s' % run_prefix)
+
+        if rev_fp is None:
+            samples.append((run_prefix, sn_by_rp[run_prefix], fwd_fp, None))
+        else:
+            rev_fn = basename(rev_fp)
+            # if we have reverse reads, make sure the matching pair also
+            # matches the run prefix:
+            if not rev_fn.startswith(run_prefix):
+                raise ValueError('Reverse read does not match this run prefix.'
+                                 '\nRun prefix: %s\nForward read: %s\n'
+                                 'Reverse read: %s\n' %
+                                 (run_prefix, fwd_fn, rev_fn))
+
+            samples.append((run_prefix, sn_by_rp[run_prefix], fwd_fp,
+                            rev_fp))
+
+        used_prefixes.add(run_prefix)
+
+    return(samples)
+
+
+def format_kneaddata_params(parameters):
+
+    params = []
+
+    for param in sorted(parameters):
+        value = parameters[param]
+
+        if value is True:
+            params.append('--%s' % param)
+            continue
```

Comment:
since these are if/elif, the `continue` statements are not needed - can you remove them?